src/main/java/org/wso2/carbon/mongodb/util/MongoDatabaseUtil.java

```diff
@@ -0,0 +1,634 @@
+     */
+    public static int getIntegerValueFromDatabase(DB dbConnection, String stmt, Map<String, Object> params)
+            throws MongoDBQueryException {
+
+        MongoPreparedStatement prepStmt = null;
+        int value = -1;
+        JSONObject jsonKeys = new JSONObject(stmt);
+        List<String> keys = getKeys(jsonKeys);
+        try {
+            prepStmt = new MongoPreparedStatementImpl(dbConnection, stmt);
+            for (String key : keys) {
+                if (!(key.equals("collection") || key.equals("projection"))) {
+                    for (Map.Entry<String, Object> entry : params.entrySet()) {
+                        if (entry.getKey().equals(key)) {
+                            if (entry.getValue() == null) {
+                                throw new MongoDBQueryException("Null Data Provided as the query parameter");
+                            } else if (entry.getValue() instanceof String) {
+                                prepStmt.setString(key, (String) entry.getValue());
+                            } else if (entry.getValue() instanceof Integer) {
+                                prepStmt.setInt(key, (Integer) entry.getValue());
+                            }
+                        }
+                    }
+                }
+            }
+            DBCursor cursor = prepStmt.find();
+            while (cursor.hasNext()) {
+                value = (int) Double.parseDouble(cursor.next().get("UM_ID").toString());
+            }
+            return value;
+        } catch (MongoDBQueryException e) {
+            throw new MongoDBQueryException("Failed to retrieve integer values from database", e);
+        } finally {
+            MongoDatabaseUtil.closeAllConnections(dbConnection, prepStmt);
+        }
+    }
+
+    /**
+     * Update user role in batch mode to database.
+     *
+     * @param dbConnection of user store
+     * @param params       values to filter from database
+     * @param stmt         query to execute in mongodb
+     * @throws MongoDBQueryException if null data provided to mongodb query or any other query error occurred
+     */
+    public static void updateUserRoleMappingInBatchMode(DB dbConnection, String stmt, Map<String, Object> params)
+            throws MongoDBQueryException {
+
+        MongoPreparedStatement prepStmt = null;
+        boolean localConnection = false;
+        JSONObject jsonKeys = new JSONObject(stmt);
+        List<String> keys = getKeys(jsonKeys);
+        try {
+            prepStmt = new MongoPreparedStatementImpl(dbConnection, stmt);
+            int batchParamIndex = -1;
+            Iterator<String> searchKeys = keys.iterator();
+            int[] values = null;
+            String listKey = "";
+            while (searchKeys.hasNext()) {
+                String key = searchKeys.next();
+                if (!(key.equals("collection") || key.equals("projection") || key.equals("$set"))) {
```

Comment:
handle null

Reply:
Resolved. Handled all the possible null pointer exceptions as you suggested in the whole project.